ticketbuyer.go

```diff
@@ -0,0 +1,233 @@
+	DisableClientTLS bool   `long:"noclienttls" description:"Disable TLS for the RPC client -- NOTE: This is only allowed if the RPC client is connecting to localhost"`
+
+	// Deprecated options which have aliases or counterparts in dcrwallet
+	AccountName       string  `long:"accountname" description:"Name of the account to buy tickets from (default: default)"`
+	TicketAddress     string  `long:"ticketaddress" description:"Address to give ticket voting rights to"`
+	PoolAddress       string  `long:"pooladdress" description:"Address to give pool fees rights to"`
+	PoolFees          float64 `long:"poolfees" description:"The pool fee base rate for a given pool as a percentage (0.01 to 100.00%)"`
+	MaxPriceAbsolute  float64 `long:"maxpriceabsolute" description:"The absolute maximum price to pay for a ticket (default: 100.0 Coin)"`
+	TxFee             float64 `long:"txfee" description:"Default regular tx fee per KB, for consolidations (default: 0.01 Coin/KB)"`
+	BalanceToMaintain float64 `long:"balancetomaintain" description:"Balance to try to maintain in the wallet"`
+}
+
+// loadTicketBuyerConfig initializes and parses the config using a config file.
+func loadTicketBuyerConfig() (*ticketBuyerConfig, error) {
+	loadConfigError := func(err error) (*ticketBuyerConfig, error) {
+		return nil, err
+	}
+
+	// Default config.
+	cfg := ticketBuyerConfig{
+		ConfigFile:        defaultTicketBuyerConfigFile,
+		MinPriceScale:     defaultMinPriceScale,
+		MaxPriceScale:     defaultMaxPriceScale,
+		AvgPriceMode:      defaultAvgPriceMode,
+		AvgPriceVWAPDelta: defaultAvgVWAPPriceDelta,
+		MaxFee:            defaultMaxFee,
+		MinFee:            defaultMinFee,
+		FeeSource:         defaultFeeSource,
+		MaxPerBlock:       defaultMaxPerBlock,
+		HighPricePenalty:  defaultHighPricePenalty,
+		BlocksToAvg:       defaultBlocksToAvg,
+		FeeTargetScaling:  defaultFeeTargetScaling,
+		MaxInMempool:      defaultMaxInMempool,
+		ExpiryDelta:       defaultExpiryDelta,
+
+		AccountName:      defaultAccountName,
+		MaxPriceAbsolute: defaultMaxPriceAbsolute,
+		TxFee:            defaultTxFee,
+	}
+
+	// A config file in the current directory takes precedence.
+	exists := false
+	if _, err := os.Stat(defaultTicketBuyerConfigFilename); !os.IsNotExist(err) {
+		exists = true
+	}
+
+	if exists {
+		cfg.ConfigFile = defaultTicketBuyerConfigFile
+	}
+
+	// Load additional config from file.
+	var configFileError error
+	parser := flags.NewParser(&cfg, flags.Default)
+	err := flags.NewIniParser(parser).ParseFile(cfg.ConfigFile)
+	if err != nil {
+		if _, ok := err.(*os.PathError); !ok {
+			fmt.Fprintln(os.Stderr, err)
+			parser.WriteHelp(os.Stderr)
+			return loadConfigError(err)
+		}
+		configFileError = err
+	}
+
+	if configFileError != nil {
+		log.Warnf("%v", configFileError)
+	}
+
+	// Check deprecated aliases.
+	if cfg.AccountName != defaultAccountName {
+		fmt.Fprintln(os.Stderr, "accountname option has been replaced by "+
```

Comment:
Since this parsing should be done after the main program config's parsing, and logging should already be set up and ready to use, print these warnings to the log.